src/Compilers/CSharp/Test/Semantic/Semantics/DelegateTypeTests.cs

```diff
@@ -382,13 +396,48 @@ public void LambdaConversions_04()
 @"using System;
 using System.Linq.Expressions;
 class Program
+{
+    static void Main()
+    {
+        LambdaExpression e = () => 1;
+        Report(e);
+        e = (LambdaExpression)(() => 2);
+        Report(e);
+    }
+    static void Report(object obj) => Console.WriteLine(obj.GetType());
+}";
+
+            var comp = CreateCompilation(source, parseOptions: TestOptions.Regular9);
+            comp.VerifyDiagnostics(
+                // (7,30): error CS1660: Cannot convert lambda expression to type 'LambdaExpression' because it is not a delegate type
+                //         LambdaExpression e = () => 1;
+                Diagnostic(ErrorCode.ERR_AnonMethToNonDel, "() => 1").WithArguments("lambda expression", "System.Linq.Expressions.LambdaExpression").WithLocation(7, 30),
```

Comment:
I wouldn't ask you to make any such change in this PR, but I am wondering if it is practical to make a change analogous to the one in this PR to improve the diagnostics when the feature isn't supported in the user's LangVersion: https://github.com/dotnet/roslyn/pull/54797/files

Reply:
[https://github.com/dotnet/roslyn/issues/56391](https://github.com/dotnet/roslyn/issues/56391)